src/commands/build.rs

```diff
@@ -9,21 +9,21 @@ use super::up::UpCommand;
 
 /// Run the build command for each component.
 #[derive(Parser, Debug)]
-#[clap(about = "Build the Spin application", allow_hyphen_values = true)]
+#[command(about = "Build the Spin application")]
 pub struct BuildCommand {
     /// Path to spin.toml.
-    #[clap(
-            name = APP_CONFIG_FILE_OPT,
-            short = 'f',
-            long = "file",
-        )]
+    #[arg(
+        short = 'f',
+        long = "file",
+        id = APP_CONFIG_FILE_OPT,
+    )]
     pub app: Option<PathBuf>,
 
     /// Run the application after building.
-    #[clap(name = BUILD_UP_OPT, short = 'u', long = "up")]
+    #[arg(id = BUILD_UP_OPT, short, long)]
     pub up: bool,
 
-    #[clap(requires = BUILD_UP_OPT)]
+    #[arg(requires = BUILD_UP_OPT, hide = true)]
```

Comment:
This is a good chance to check in on what we'd like to see re: these up args.

Previous to this PR, we have:

![spin-build-old](https://user-images.githubusercontent.com/9322017/215620142-d3890756-c5a1-4483-bb9f-74de52d95f9c.jpg)

With this PR, there is no mention of up args:

<img width="500" alt="Screenshot 2023-01-30 at 4 26 12 PM" src="https://user-images.githubusercontent.com/9322017/215619959-1f8e6639-28bf-4662-bab0-9f7819ee3ec0.png">

Admittedly, the old style of just showing 'UP_ARGS' is perhaps not ideal (also b/c `spin up` technically takes options/flags and not args).  But I do want to check with @itowlson to get input here.

As an aside, it would be nifty to inject all of the valid/applicable spin up flags/opts here -- I wonder if we can do so via clap?  Regardless, this would/should be a follow-up feature, to keep this PR focused purely on updating clap usage.

Reply:
It would be good to show _something_ to indicate that `up` options can be provided if `--up` is present, though if the worst comes to the worst we can mention it in `--up` doc string instead.

Reply:
I started looking into a better way to display it and forgot to leave the change out for now.

I don't know if there's a right way to do this. I guess the flag is trying to do too much?

Reply:
Thanks @cardoso; sticking with current behavior is great for now.  I'll create a separate issue for potentially improving things here.